contracts/contracts_test.go

```diff
@@ -1,130 +1,30 @@
-package contracts
+package contracts_test
 
 import (
-	"context"
-	"fmt"
-	// "math/big"
+	"encoding/json"
 	"testing"
-	"time"
 
-	"github.com/joho/godotenv"
-	// "github.com/NethermindEth/juno/core/felt"
-	// starknetgo "github.com/NethermindEth/starknet.go"
-	// "github.com/NethermindEth/starknet.go/account"
 	"github.com/NethermindEth/starknet.go/artifacts"
-	// devtest "github.com/NethermindEth/starknet.go/test"
-	// "github.com/NethermindEth/starknet.go/types"
-	// "github.com/NethermindEth/starknet.go/utils"
+	"github.com/NethermindEth/starknet.go/contracts"
+	"github.com/NethermindEth/starknet.go/rpc"
+	"github.com/test-go/testify/require"
 )
 
-func TestRPC_InstallCounter(t *testing.T) {
-	godotenv.Load()
-	testConfiguration := beforeEach(t)
-
-	type TestCase struct {
-		providerType  ProviderType
-		CompiledClass []byte
-		Salt          string
-		Inputs        []string
-	}
-
-	TestCases := map[string][]TestCase{
-		"devnet": {
-			{
-				providerType:  ProviderRPC,
-				CompiledClass: artifacts.CounterCompiled,
-				Salt:          "0x01",
-				Inputs:        []string{},
-			},
-		},
-	}[testEnv]
-	for _, test := range TestCases {
-		ctx := context.Background()
-		ctx, cancel := context.WithTimeout(ctx, time.Second*60)
-		defer cancel()
-		var err error
-		var tx *DeployOutput
-		switch test.providerType {
-		case ProviderRPC:
-			provider := RPCProvider(*testConfiguration.rpc)
-			tx, err = provider.deployAndWaitWithWallet(ctx, test.CompiledClass, test.Salt, test.Inputs)
-		default:
-			t.Fatal("unsupported client type", test.providerType)
-		}
-		if err != nil {
-			t.Fatal("should succeed, instead", err)
-		}
-		fmt.Println("deployment transaction", tx.TransactionHash)
-	}
+func TestUnmarshalContractClass(t *testing.T) {
+	compiledClass := artifacts.ExampleWorldSierra
+	var class rpc.ContractClass
+	err := json.Unmarshal(compiledClass, &class)
+	require.NoError(t, err)
+	require.Equal(t, class.SierraProgram[0].String(), "0x1")
+	require.Equal(t, class.SierraProgram[1].String(), "0x3")
```

Comment:
Consider changing `require.Equal(...)` to `assert.Equal(...)` because `require` functions stop execution immediately if arguments are not equal, but `assert` will keep going. It's helpful to see in one run which fields are correct and which are not. 

In codebases usually `require.NoError(...)` is used instead of `assert.NoError(...)` because most likely other arguments will be nil and there is no point to keep going (facing nil pointer panic)